integrations/examples/calling-guest-functions.md

```diff
@@ -0,0 +1,273 @@
+---
+description: >-
+  A Wasm module can export entities, like functions, memories, globals and
+  tables. This example illustrates how to call exported functions.
+---
+
+# Calling guest functions
+
+In this example we'll see how to use exported functions.
+
+Exported function are the entities you will probably use the most: they will be your entrypoint to calling Wasm module logic.
+
+Exported function come in two flavors:
+
+* Dynamic functions;
+* Native functions.
+
+We'll cover both flavors in this example.
+
+First we are going to want to initialize a new project. To do this we can navigate to our project folder, or create one. In this example, we will create a new project. Lets create it and navigate to it:
+
+{% tabs %}
+{% tab title="Rust" %}
+{% hint style="info" %}
+The final code for this example can be found on [GitHub](https://github.com/wasmerio/wasmer/blob/master/examples/exports_function.rs).
+
+_Please take a look at the_ [_setup steps for Rust_](../rust/setup.md)_._
+{% endhint %}
+
+```bash
+cargo new exports-function
+cd exports-function
+```
+
+This should generate two important files for us, `Cargo.toml` and `src/main.rs`. The `Cargo.toml` is a file that describes your project and its dependencies. The `src/main.rs` is the entry point for your project, and contains the `fn main() { .. }` that is run when the project is executed.
+
+We then modify the `Cargo.toml` to add the Wasmer dependencies as shown below:
+
+{% code title="Cargo.toml" %}
+```rust
+[package]
+name = "exports-function"
+version = "0.1.0"
+authors = ["The Wasmer Engineering Team <engineering@wasmer.io>"]
+edition = "2018"
+
+[dependencies]
+# The Wasmer API
+wasmer = "1.0.0-alpha4"
+```
+{% endcode %}
+{% endtab %}
+
+{% tab title="Go" %}
+{% hint style="info" %}
+The final code for this example can be found on [GitHub](https://github.com/wasmerio/wasmer-go/blob/master/examples/example_exports_function_test.go).
+
+_Please take a look at the_ [_setup steps for Go_](../go/setup.md)_._
+{% endhint %}
+
+```text
+mkdir wasmer-example-exports-function
+cd wasmer-example-exports-function
+go mod init github.com/$USER/wasmer-example-exports-function
+```
+{% endtab %}
+
+{% tab title="C/C++" %}
+{% hint style="info" %}
+The final code for this example can be found on [GitHub](https://github.com/wasmerio/wasmer/blob/master/lib/c-api/examples/instance.c).
+
+_Please take a look at the_ [_setup steps for C/C++_](../c/setup.md)_._
+{% endhint %}
+
+```text
+mkdir wasmer-example-exports-function
+cd wasmer-example-exports-function
+vim Makefile
+```
+
+Let's create a simple `Makefile`:
+
+```c
+CFLAGS = -g -I$(WASMER_C_API)/include
+LDFLAGS = -L$(WASMER_C_API)/lib -Wl,-rpath,$(WASMER_C_API)/lib
```

Comment:
We have the following commands that we can use, please try them out and choose them accordingly:

```
$ wasmer config --cflags
-I/Users/syrus/.wasmer/include/wasmer
```

```
$ wasmer config --includedir
/Users/syrus/.wasmer/include
```

```
$ wasmer config --libdir
/Users/syrus/.wasmer/lib
```

```
$ wasmer config --libs
-L/Users/syrus/.wasmer/lib -lwasmer
```

Reply:
way better than what I actually did! will fix that